compiler/seminst.nim

```diff
@@ -319,6 +319,47 @@ proc instantiateProcType(c: PContext, pt: TIdTable,
   prc.typ = result
   popInfoContext(c.config)
 
+{.emit: "NIM_EXTERNC".}
```

Comment:
What?

Reply:
for `nim cpp` to work with exportc, best workaround I know of to https://github.com/nim-lang/Nim/issues/10578#issuecomment-461635978
The alternative is move around a bunch of declarations until it compiles, which seems more intrusive.

Reply:
Bootstrapping works with ``nim cpp``, it's covered by the CIs. I don't understand why this line of code is necessary.

Reply:
without it, 
`nim cpp -o:bin/nim_temp -d:release compiler/nim.nim`
gives:
```
Undefined symbols for architecture x86_64:
  "_generateInstanceEnableIf", referenced from:
```
because it uses C++ instead of C mangling, as mentioned in  #10578 :
```
nm /Users/timothee/.cache/nim/nim_r/*.cpp.o | grep generateInstanceEnableIf
0000000000006560 T __Z24generateInstanceEnableIfP41tyObject_TContext_C9bB2okDBl3OKGUBiY5k5UgP36tyObject_TSym_JpsEh5i1AcKChGYbg7aV4w43tyObject_TIdTable_9aiv9bp2t5icFNINcg9c4xi1Q41tyObject_TLineInfo_WGx4rAexNXnczy1Avn646QP38tyObject_TNode_bROa11lyF5vxEN9aYNbHmhwb
                 U _generateInstanceEnableIf
```

feel free to try it out

Reply:
It's not that I don't believe you, I thought you could speed up my understanding. How can bug #10578 ever be triggered by plain old simple pure Nim compiler development?

Reply:
maybe you didn't see the `{.exportc.}` in `generateInstanceEnableIf` (see https://github.com/nim-lang/Nim/pull/12048/files/661392c37f9bffa1e1473014ac9f8cfdb5f9fe9e#r320858573 for rationale)
I've just sent out https://github.com/nim-lang/Nim/pull/12144 which fixes #10578 and obviates need for `{.emit: "NIM_EXTERNC".}`

Reply:
now that #12144 was merged I've removed `{.emit: "NIM_EXTERNC".}`